src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DraftReferralService.kt

```diff
@@ -484,6 +508,108 @@ class DraftReferralService(
     }
   }
 
+  private fun createProbationPractitionerDetails(draftReferral: DraftReferral, referral: Referral) {
+    if (saveProbationPractitionerDetails) {
+      referral.probationPractitionerDetails = probationPractitionerDetailsRepository.save(
+        ProbationPractitionerDetails(
+          id = UUID.randomUUID(),
+          referral = referral,
+          nDeliusName = draftReferral.nDeliusPPName,
+          nDeliusEmailAddress = draftReferral.nDeliusPPEmailAddress,
+          nDeliusPDU = draftReferral.nDeliusPPPDU,
+          name = draftReferral.ppName,
+          emailAddress = draftReferral.ppEmailAddress,
+          pdu = draftReferral.ppPdu,
+          probationOffice = draftReferral.ppProbationOffice!!,
+        ),
+      )
+      referralRepository.save(referral)
+      checkAndMeasureDeliusData(draftReferral)
+    }
+  }
+
+  private fun checkAndMeasureDeliusData(draftReferral: DraftReferral) {
+    val eventName = "probationPractitionerDetailsLookUp"
+    when (draftReferral.hasValidDeliusPPDetails) {
+      true -> telemetryClient.trackEvent(
+        eventName,
+        mapOf(
+          "result" to "valid delius details present",
+          "referralId" to draftReferral.id.toString(),
+        ),
+        null,
+      )
+      false -> {
+        telemetryClient.trackEvent(
+          eventName,
+          mapOf(
+            "result" to "valid delius details not present",
```

Comment:
Could the result be set to "invalid delius details" here? When I first read this, I thought this meant the delius details are valid but do not exist. And if hasValidDeliusPPDetails is true, we could probably also omit 'present' on line 537?

Reply:
Agreed and done